presto-hive/src/test/java/io/prestosql/plugin/hive/AbstractTestHive.java

```diff
@@ -4552,6 +4631,9 @@ protected static void assertPageSourceType(ConnectorPageSource pageSource, HiveS
         if (pageSource instanceof RecordPageSource) {
             RecordCursor hiveRecordCursor = ((RecordPageSource) pageSource).getCursor();
             hiveRecordCursor = ((HiveRecordCursor) hiveRecordCursor).getRegularColumnRecordCursor();
+            if (hiveRecordCursor instanceof HiveBucketValidationRecordCursor) {
+                hiveRecordCursor = ((HiveBucketValidationRecordCursor) hiveRecordCursor).delegate();
+            }
```

Comment:
How does HiveBucketValidationRecordCursor play with HiveCoercionRecordCursor?
are they applied in the right order? would be good to have a test

same applies to the case when reading with a page source

Reply:
It shouldn't matter, since the code in `HiveSplitManager` already validates that the partition bucketing is compatible:

* same = no coercion
* compatible = adaptation (does validation itself)
* incompatible = query fails

Any coercion would be for non-bucketing columns.